File: case/Offer_18_删除链表的节点.go
package _case

/**
https://leetcode.cn/problems/shan-chu-lian-biao-de-jie-dian-lcof/?envType=study-plan-v2&envId=coding-interviews
*/

func deleteNode(head *ListNode, val int) *ListNode {
	root := &ListNode{Next: head}
	roo2 := root
	for root != nil && root.Next != nil {
		if root.Next.Val == val {
			root.Next = root.Next.Next
		}

		root = root.Next
	}

	return roo2.Next
}

File: case/Offer_18_删除链表的节点.js
// https://leetcode.cn/problems/shan-chu-lian-biao-de-jie-dian-lcof/?envType=study-plan-v2&envId=coding-interviews

var deleteNode = function(head, val) {
    const root = new ListNode();
    root.next = head;
    let cur = root
    while (cur && cur.next) {
        if (cur.next.val === val) {
            cur.next = cur.next.next;
        }
        cur = cur.next;
    }
    return root.next;
};